### PR TITLE
Skip inspect of built-in networks on stack deploy

### PR DESCRIPTION
**Pull request by alexmavr**
**- What I did**

The current flow of `docker stack deploy -c` performs a network inspect operation for every external network. However, built-in networks are always known to exist so these inspect calls are redundant. This PR removes them.

**- How I did it**
For every external network in a compose file, omit the network inspect operation if it's a built-in network ("none", "default", "container", "bridge" or "host")

**- How to verify it**
- Use the provided unit test

or 

- Build a docker API server that returns a 404 for the "host" and "bridge" networks (such as https://github.com/docker/swarm) and run the following compose file:

```
version: '3.2'
services:
  api:
    image: nginx
    networks:
      host : null
networks:
  host:
    external: true
```

Then, verify the deployed task is attached to the "host" network on the target node.

**- Description for the changelog**

* Stack deploy for compose files no longer inspects built-in networks

**- A picture of a cute animal (not mandatory but encouraged)**

important-stuff

Signed-off-by: Alex Mavrogiannis <alex.mavrogiannis@docker.com>

### CODE REVIEW

**Comment by alexmavr** [comment]
cc @mavenugo

**Comment by mavenugo** [comment]
LGTM

Not sure why the CI is failing though...

**Comment by alexmavr** [comment]
Amended my commit with no code diff - looks like CI just needed a little nudge

File: cli/command/stack/deploy_composefile_test.go [hunk 1]
```diff
@@ -56,7 +58,8 @@ func TestValidateExternalNetworks(t *testing.T) {
 			expectedMsg:  "Unexpected",
 		},
 		{
-			network: "host",
+			noInspect: true,
+			network:   "host",
```
Comment (dnephin) [inline cli/command/stack/deploy_composefile_test.go]: I'm not rally a fan of this storing state in the testcase struct.

I think the same thing can be accomplished by setting `inspectError`, and asserting there was no error, right?

**Comment by thaJeztah** [comment]
This didn't make the cut for 17.07, opened https://github.com/docker/docker-ce/issues/177 to back port it

**Pull request merged**